keda/templates/metrics-server/clusterrolebinding.yaml

```diff
@@ -37,9 +38,10 @@ roleRef:
   name: extension-apiserver-authentication-reader
 subjects:
 - kind: ServiceAccount
-  name: {{ .Values.serviceAccount.name }}
+  name: {{ (.Values.serviceAccount.metricServer).name | default .Values.serviceAccount.name }}
   namespace: {{ .Release.Namespace }}
 ---
+# todo this
```

Comment:
Currently, there is just one service account for all 3 deployments that's configurable by `.Values.serviceAccount.*`. I am splitting this to individual 3 service accounts that can have their own settings in the values.yaml. 
However, to stay backward compatible, I guess we should allow users to use their old values.yaml w/ new helm chart version, so I choose this "fallback" approach. Where it first tries the new property and if it's not there (user didn't updated their values.yaml yet) it fall backs to the old path and at the same time he/she will be notified in the `NOTES.txt` that they are using a deprecated api.

I agree, It's kind of ugly. I was thinking having it there for some transition period, where the deprecation is in place and then remove the fallback parts (`(.Values.serviceAccount.metricServer).name | default <this>`)

Reply:
ah, I missed the `todo comment`, right I put it there, because I wasn't sure about this one. But let's rather keep it there, because on Kind and k3d the service account actually really exist and on other k8s distros it won't hurt, the rollbinding can be created even if the SA is not there. So I am removing the "todo" comment